lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java

```diff
@@ -212,6 +261,40 @@ public void onXMLContent(ICompletionRequest request, ICompletionResponse respons
 		}
 	}
 
+	private InsertionStrategy computeGAVInsertionStrategy(ICompletionRequest request) {
+		if (request.getParentElement() == null) {
+			return null;
+		}
+		switch (request.getParentElement().getLocalName()) {
+		case "dependencies": return new InsertionStrategy.NodeWithChildrenInsertionStrategy("dependency");
+		case "dependency": return InsertionStrategy.CHILDREN_ELEMENTS;
+		case "plugins": return new InsertionStrategy.NodeWithChildrenInsertionStrategy("plugin");
+		case "plugin": return InsertionStrategy.CHILDREN_ELEMENTS;
+		case "artifactId": return InsertionStrategy.ELEMENT_VALUE_AND_SIBLING;
+		}
+		return InsertionStrategy.ELEMENT_VALUE_AND_SIBLING;
+	}
+
+	private Optional<MavenProject> computeFilesystemParent(ICompletionRequest request) {
+		Optional<String> relativePath = null;
+		if (request.getParentElement().getLocalName().equals("parent")) {
+			relativePath = DOMUtils.findChildElementText(request.getNode(), "relativePath");
+		} else {
+			relativePath = DOMUtils.findChildElementText(request.getParentElement().getParentElement(), "relativePath");
+		}
+		if (!relativePath.isPresent()) {
+			relativePath = Optional.of("..");
+		}
+		File referencedTargetPomFile = new File(new File(URI.create(request.getXMLDocument().getTextDocument().getUri())).getParentFile(), relativePath.orElse(""));
```

Comment:
Just to make sure I follow:
1. We check if the <relativePath> was writen by the user
2. We then create a grandparent file of the current XML document (which is most likely a directory) or create a file using the relative path
3. If the file created in step 2 was a directory, create a File object for a pom.xml in that directory (or if it's already a file then move to step 4)
4. Parse the file object and return it

Reply:
If relativePath is not set, we assume it's `..` (the parent folder).